packages/react/src/experimental/OneDataCollection/useSelectable.ts

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useMemo, useState } from "react"
+import type { FiltersDefinition } from "./Filters/types"
+import type { SortingsDefinition } from "./sortings"
+import { DataSource, PaginationInfo, RecordType } from "./types"
+
+export function useSelectable<
+  Record extends RecordType,
+  Filters extends FiltersDefinition,
+  Sortings extends SortingsDefinition,
+>(
+  data: ReadonlyArray<Record>,
+  paginationInfo: PaginationInfo | null,
+  source: DataSource<Record, Filters, Sortings>,
+  onSelectItems?: (allSelected: boolean, items: ReadonlyArray<Record>) => void
+): {
+  isAllSelected: boolean
+  selectedItems: Map<number | string, Record>
+  isPartiallySelected: boolean
+  handleSelectItemChange: (item: Record, checked: boolean) => void
+  handleSelectAll: (checked: boolean) => void
+} {
+  // itemsState is the state of the selected items
+  const [itemsState, setItemsState] = useState<
+    Map<number | string, { item: Record; checked: boolean }>
+  >(new Map())
+
+  const [selectedItems, unselectedItems] = useMemo(() => {
+    const selected = new Map()
+    const unselected = new Map()
+    itemsState.forEach((value, id) => {
+      if (value.checked) {
+        selected.set(id, value.item)
+      } else {
+        unselected.set(id, value.item)
+      }
+    })
+    return [selected, unselected]
+  }, [itemsState])
+
+  const selectedCount: number = selectedItems.size
+  const unselectedCount: number = unselectedItems.size
```

Comment:
If I'm not wrong, there is no need to type variables here:
```suggestion
  const selectedCount = selectedItems.size
  const unselectedCount = unselectedItems.size
```